components/productions/index/event.vue

```diff
@@ -48,40 +48,48 @@
 </template>
 
 <script>
-  import { format } from 'date-fns';
-
   export default {
     props: {
-      event: {
-        type: Object,
-        default: undefined,
+      id: {
+        type: String,
+        default: '',
+      },
+      name: {
+        type: String,
+        default: '',
+      },
+      location: {
+        type: String,
+        default: '',
       },
     },
     data() {
       return {
         isDetailVisible: false,
+        period: '',
       };
     },
     computed: {
       locale() {
         return this.$i18n.locale;
       },
-      period() {
-        return `${this.parseDate(this.event.startDate)} - ${this.parseDate(
-          this.event.endDate,
-        )}`;
-      },
+    },
+    async created() {
+      this.period = await this.getPeriod();
     },
     methods: {
       handleClickToggleShowDetail() {
         this.isDetailVisible = !this.isDetailVisible;
       },
-      parseDate(date) {
-        return format(new Date(date), 'dd/MM/yyyy');
-      },
       handleClickDelete(eventId) {
         this.$emit('clickDelete', eventId);
       },
+      async getPeriod() {
+        return await this.$api.events.getCalendarSummary({
+          id: this.id,
+          locale: this.locale,
+        });
+      },
     },
```

Comment:
can't we just call the api helper directly in created? This just adds an extra layer of abstraction